### PR TITLE
[WIP][SHARE-632][Feature] Create mods harvester and transformer

### PR DESCRIPTION
**Pull request by leb2dg**
## Purpose
Add MODS harvester and transformer to support TDAR and others.

### CODE REVIEW

File: share/normalize/mods.py [hunk 1]
```diff
@@ -0,0 +1,431 @@
...
+
+from share.normalize import ctx, tools
+from share.normalize.parsers import Parser
+from share.normalize.normalizer import Normalizer
+
+
+logger = logging.getLogger(__name__)
+
+
+def force_text(data):
```
Comment (chrisseto) [inline share/normalize/mods.py]: Double indent

File: share/normalize/mods.py [hunk 2]
```diff
@@ -0,0 +1,431 @@
...
+
+        # (dc:title) http://www.loc.gov/standards/mods/userguide/titleinfo.html
+        titleInfo = tools.Try(ctx['mods:titleInfo'])
+
+        # Extra
+        # (dc:type) http://www.loc.gov/standards/mods/userguide/typeofresource.html
+        typeOfResource = tools.Try(ctx['mods:typeOfResource'])
+
+    def check_status(self, status):
+        if status == 'deleted':
```
Comment (chrisseto) [inline share/normalize/mods.py]: you can just return `status == 'deleted'`

File: share/normalize/mods.py [hunk 3]
```diff
@@ -0,0 +1,431 @@
...
+            'mods:recordInfo',
+            'mods:relatedItem',
+            'mods:subject',
+            'mods:tableOfContents',
+            'mods:targetAudience',
+            'mods:titleInfo',
+            'mods:typeOfResource'
+        ]
+
+        for field in top_level_fields:
```
Comment (chrisseto) [inline share/normalize/mods.py]: Could this just be `flattened_data = {**unwrapped_data['record']['metadata']['mods:mods']}`

File: share/normalize/mods.py [hunk 4]
```diff
@@ -0,0 +1,431 @@
...
+
+    title = tools.Join(tools.RunPython(force_text, tools.Try(ctx['mods:titleInfo']['mods:title'])))
+
+    # TODO
+    # This attribute is used with shareable="no" for data that may be proprietary or is rights
+    # protected and should not be used outside of a local system (such as providing to harvesters).
+    description = tools.Join(tools.RunPython(force_text, tools.Try(ctx['mods:abstract'])))
+
+    # do we want isbns?
+    identifiers = tools.Map(
```
Comment (chrisseto) [inline share/normalize/mods.py]: Bit of an odd one, yes if its in a book but not a book section.

Probably fine to collect for the moment.

File: share/normalize/mods.py [hunk 5]
```diff
@@ -0,0 +1,431 @@
...
+                fixed.append(datum['#text'])
+            elif isinstance(datum, str):
+                fixed.append(datum)
+            else:
+                raise Exception(datum)
+        return fixed
+
+
+def not_invalid(identifier):
+    if 'invalid' in identifier:
```
Comment (chrisseto) [inline share/normalize/mods.py]: `return 'invalid' in identifier`

**Comment by aaxelb** [comment]
I continued this work and rebased onto #596. I'll open a PR once #596 is merged, but for now it can be reviewed at aaxelb/SHARE#2

**Comment by aaxelb** [comment]
Subsumed by #602

**Pull request closed without merging**